src/oss-diff/DiffTool.cs

```diff
@@ -0,0 +1,207 @@
+            public IEnumerable<string> Targets { get; set; } = Array.Empty<string>();
+
+            
+        }
+        static async Task Main(string[] args)
+        {
+            var diffTool = new DiffTool();
+            await diffTool.ParseOptions<Options>(args).WithParsedAsync<Options>(diffTool.RunAsync);
+        }
+
+        public async Task RunAsync(Options options)
+        {
+            if (options.Targets.Count() != 2)
+            {
+                Logger.Error("Must provide exactly two packages to diff.");
+                return;
+            }
+
+            if (options.Context > 0)
+            {
+                options.Before = options.Context;
+                options.After = options.Context;
+            }
+
+            (PackageURL purl1, PackageURL purl2) = (new PackageURL(options.Targets.First()), new PackageURL(options.Targets.Last()));
+            var manager = ProjectManagerFactory.CreateProjectManager(purl1, options.DownloadDirectory ?? Path.GetTempPath());
+            var manager2 = ProjectManagerFactory.CreateProjectManager(purl2, options.DownloadDirectory ?? Path.GetTempPath());
+
+            if (manager is not null && manager2 is not null)
+            {
+                var locations = await manager.DownloadVersion(purl1, true, options.UseCache);
+
+                var locations2 = await manager2.DownloadVersion(purl2, true, options.UseCache);
+
+                Dictionary<string, (string, string)> files = new Dictionary<string, (string, string)>();
+                foreach (var directory in locations)
+                {
+                    foreach (var file in Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories))
+                    {
+                        var contents = File.ReadAllText(file);
+                        files.Add(string.Join(Path.DirectorySeparatorChar,file.Substring(directory.Length).Split(Path.DirectorySeparatorChar)[2..]), (contents, string.Empty));
+                    }
+                }
+
+                foreach (var directory in locations2)
+                {
+                    foreach (var file in Directory.EnumerateFiles(directory, "*", SearchOption.AllDirectories))
+                    {
+                        var contents = File.ReadAllText(file);
+                        var key = string.Join(Path.DirectorySeparatorChar, file.Substring(directory.Length).Split(Path.DirectorySeparatorChar)[2..]);
+
+                        if (files.ContainsKey(key))
+                        {
+                            var existing = files[key];
+                            existing.Item2 = contents;
+                            files[key] = existing;
+                        }
+                        else
+                        {
+                            files[key] = (string.Empty, contents);
+                        }
+                    }
+                }
+
+                foreach (var filePair in files)
```

Comment:
Ill create some feature requests for this, including recognizing directory offsetting, recursive extraction of content packages, json/sarif support etc. (for example, if color output is run in a container (toolshed), the output will come out garbled.

Reply:
The library I'm using respects NO_COLOR already per Mike's request.

Reply:
But for the other ones definitely create some feature requests

Reply:
I added sarif and file location output options.

What do you mean by directory offsetting?

Do you want me to run all the found files through recursive extractor?

Reply:
Recursive extractor bit ended up easier than I anticipated. Now included here.

Reply:
awesome with the sarif and location output, I'll go through those. 
By directory offsetting I mean that the directory was moved. If we have a/b/c moved to d/e/c, how do we figure out if that happened? Also if you are just comparing the file names, a/b/a.txt and c/e/a.txt might be paired up, but not be related at all. 
For recursive extractor, maybe test the file for archive (does recursive extractor have a fast function that tells you if its a valid/well known archive or not?), and extract it if it is. Does not need to be part of the PR, its another feature request. looks like you have already done that part...will go through the changes.

Reply:
> By directory offsetting I mean that the directory was moved. If we have a/b/c moved to d/e/c, how do we figure out if that happened?

This is too large to include in this PR.  I'm not sure how important it is.  Refactors that move lots of files around are pretty rare.

> Also if you are just comparing the file names, a/b/a.txt and c/e/a.txt might be paired up, but not be related at all.

I compare the full relative path names.

> For recursive extractor, maybe test the file for archive (does recursive extractor have a fast function that tells you if its a valid/well known archive or not?), and extract it if it is.

This is already done and is how RecursiveExtractor works if you ask it to extract.  If you ask it to extract a non archive it just gives you the content back in a FileEntry object.